src/management-system-v2/app/(dashboard)/processes/[processId]/milestone-selection-section.tsx

```diff
@@ -37,119 +129,167 @@ const MilestoneModal: React.FC<MilestoneModalProperties> = ({ show, close }) =>
+  };
 
-    // remove milestones that do not exist anymore
-    milestones.forEach((oldMilestone) => {
-      if (!newMilestones.find((milestone) => milestone.id === oldMilestone.id)) {
-        setProceedElement(selectedElement.businessObject, 'Milestone', null, {
-          id: oldMilestone.id,
-        });
-      }
+  const openMilestoneModal = (initialMilestoneValues: {
+    id: string;
+    name: string;
+    description?: string;
+  }) => {
+    setInitialMilestoneValues(initialMilestoneValues);
+    setIsMilestoneModalOpen(true);
+  };
+
+  const addMilestone = (newMilestone: { id: string; name: string; description?: string }) => {
+    const modeling = modeler!.getModeling();
+    setProceedElement(selectedElement.businessObject, 'Milestone', undefined, newMilestone);
+    modeling.updateProperties(selectedElement as any, {
+      extensionElements: selectedElement.businessObject.extensionElements,
     });
+  };
 
+  const removeMilestone = (milestoneId: string) => {
+    const modeling = modeler!.getModeling();
+    setProceedElement(selectedElement.businessObject, 'Milestone', null, {
+      id: milestoneId,
+    });
     modeling.updateProperties(selectedElement as any, {
       extensionElements: selectedElement.businessObject.extensionElements,
     });
   };
 
   return (
     <>
-      <Space direction="vertical" size="large" style={{ width: '100%' }}>
-        <b>Milestones</b>
-        <Select
-          style={{ width: '100%' }}
-          mode="multiple"
-          fieldNames={{ label: 'name', value: 'id' }}
-          options={milestones}
-          value={milestones.map((milestone) => ({ value: milestone.id, label: milestone.name }))}
-          allowClear
-          placeholder="Select Milestones"
-          onChange={(_, selectedMilestones) => {
-            updateMilestones(
-              Array.isArray(selectedMilestones) ? selectedMilestones : [selectedMilestones],
-            );
-          }}
-          dropdownRender={(menu) => (
-            <>
-              {menu}
-              <Divider style={{ margin: '8px 0' }} />
-              <Space style={{ padding: '0 8px 4px', display: 'flex', justifyContent: 'center' }}>
-                <Button
-                  type="text"
-                  icon={<PlusOutlined />}
-                  onClick={() => setIsMilestoneModalOpen(true)}
-                >
-                  Create new Milestone
-                </Button>
-              </Space>
-            </>
-          )}
-        ></Select>
+      <Space direction="vertical" style={{ width: '100%' }}>
+        <Divider style={{ display: 'flex', alignItems: 'center', fontSize: '0.85rem' }}>
+          <span style={{ marginRight: '0.3em' }}>Milestones</span>
+          <PlusOutlined
+            onClick={() => {
+              setIsMilestoneModalOpen(true);
+            }}
+          ></PlusOutlined>
+        </Divider>
+        <Table
+          pagination={{ pageSize: 5 }}
+          columns={[
```

Comment:
There is currently an error displayed in the browser console if there are milestones in the table. This seems to happen because there are no ids defined for the data rows. Using `rowKey='id'` should fix this

Reply:
Done.